docs/rag/adrs/adr-001-vectordb.md

```diff
@@ -0,0 +1,26 @@
+# ADR 001: Initial InstructLab Vector Store
+
+## Context
+
+One of the first choices to make in implementing RAG is to choose an initial vector store to develop against. Though the usage of frameworks like LangChain or Haystack make it easy to swap vector databases, we need a working end to end implementation for RAG that is tested against and available to install with InstructLab. There are many options (see [here](https://docs.haystack.deepset.ai/docs/choosing-a-document-store)). 
+
+Our main long-term requirements are that our chosen store have fully-developed document update (and thus some sort of notion of primary key), that it be scalable to cluster size, and that it have a permission live (Apache, MIT, or similar). Among the available choices, [Milvus](https://milvus.io/) provides strategic advantage due to its [investment from watsonx](https://www.ibm.com/new/announcements/ibm-watsonx-data-vector-database-ai-ready-data-management).
+
+Milvus can be used in-process ([Milvus Lite](https://milvus.io/docs/milvus_lite.md)), single-node ([Milvus](https://milvus.io/docs/prerequisite-docker.md)), or cluster-scale ([Milvus Distributed](https://milvus.io/docs/prerequisite-helm.md)).
+
+## Decision
+
+InstructLab will initially integrate with and use Milvus Lite for vector storage and retrieval augmented generation.
+
+## Status
+
+Accepted
+
+## Consequences
```

Comment:
I think we need something in here about dependencies of Milvus and their licensing terms because I know there have been a lot of complaints about that.  However, I don't remember the details so let's try to run them down offline, ASAP.

Reply:
I tried to address the license issue in some way in the `Context` section. As for the dependencies issue, my perspective when writing this was that those issues are always present and are not specific to this decision, so there is no actual practical reason to call them out further.